onnxruntime/contrib_ops/cpu/transformers/beam_search_parameters.cc

```diff
@@ -30,25 +32,32 @@ void BeamSearchParameters::ParseFromInputs(OpKernelContext* context) {
   const auto& dims = input_ids->Shape().GetDims();
   ORT_ENFORCE(dims.size() == 2, "input_ids shall have 2 dimensions. Got ", dims.size());
   batch_size = static_cast<int>(dims[0]);
-  sequence_length = static_cast<int>(dims[1]);
+
+  // For T5, output sequence starts with decoder_start_token_id, so its sequence length is 1
+  sequence_length = this->decoder_start_token_id >= 1 ? 0 : static_cast<int>(dims[1]);
```

Comment:
Is it better to use model type to set the seq_len?

Reply:
yes. I'll update it.